qiskit/transpiler/passes/optimization/optimize_1q_gates.py

```diff
@@ -145,7 +146,7 @@ def run(self, dag):
                 # exact and approximate rewriting.
 
                 # Y rotation is 0 mod 2*pi, so the gate is a u1
-                if np.mod(right_parameters[0], (2 * np.pi)) == 0 \
+                if np.mod(np.round(right_parameters[0], _DECIMAL_ROUND), (2 * np.pi)) == 0 \
```

Comment:
Rounding before taking the mod seems a little unusual but I couldn't find a case (except for huge theta) where this would changes the outcome. `abs(right_parameters[0] - 2*pi*(right_parameters[0]//(2*pi)) < _CHOP_THRESHOLD` would be an alternative, but is more verbose.

Reply:
In  04ec2fb06I moved the round out the mod.